log/handlers.ts

```diff
@@ -33,21 +57,27 @@ export class ConsoleHandler extends BaseHandler {
 
 export abstract class WriterHandler extends BaseHandler {
   protected _writer: Writer;
+  private _encoder = new TextEncoder();
 
   log(msg: string) {
-    const encoder = new TextEncoder();
-    // promise is intentionally not awaited
-    this._writer.write(encoder.encode(msg + "\n"));
+    this._writer.write(this._encoder.encode(msg + "\n"));
   }
 }
 
+interface FileHandlerOptions extends HandlerOptions {
+  filename: string;
+}
+
 export class FileHandler extends WriterHandler {
   private _file: File;
   private _filename: string;
 
-  constructor(levelName: string, filename: string) {
-    super(levelName);
-    this._filename = filename;
+  constructor(
+    levelName: string, 
+    options: FileHandlerOptions = { filename: "log.txt" }
```

Comment:
this is just placeholder filename, I want to know your opinion

Reply:
Not so happy with having an options interface with only one item in it. (cc @piscisaureus)
It seems if you're constructing a `FileHandler`, specifying a filename should be part of the process - i.e. no default value.

Reply:
Actually it's two items, because `FileHandlerOptions` has `filename` and `formatter`. +1 for no default value